smarc_bt/src/mission_plan.py

```diff
@@ -473,6 +539,212 @@ def path_to_list(self, path_msg):
+        """
+
+        (p1x, p1y), (p2x, p2y), (cx, cy) = pt1, pt2, circle_center
+        (x1, y1), (x2, y2) = (p1x - cx, p1y - cy), (p2x - cx, p2y - cy)
+        dx, dy = (x2 - x1), (y2 - y1)
+        dr = (dx ** 2.0 + dy ** 2.0)**.5
+        big_d = x1 * y2 - x2 * y1
+        discriminant = circle_radius ** 2.0 * dr ** 2.0 - big_d ** 2.0
+
+        if discriminant < 0:  # No intersection between circle and line
+            return []
+        else:  # There may be 0, 1, or 2 intersections with the segment
+            intersections = [
+                (cx + (big_d * dy + sign * (-1 if dy < 0 else 1) * dx * discriminant**.5) / dr ** 2.0,
+                 cy + (-big_d * dx + sign * abs(dy) * discriminant**.5) / dr ** 2.0)
+                for sign in ((1, -1) if dy < 0 else (-1, 1))]  # This makes sure the order along the segment is correct
+            if not full_line:  # If only considering the segment, filter out intersections that do not fall within the segment
+                fraction_along_segment = [(xi - p1x) / dx if abs(dx) > abs(dy) else (yi - p1y) / dy for xi, yi in intersections]
+                intersections = [pt for pt, frac in zip(intersections, fraction_along_segment) if 0 <= frac <= 1]
+            if len(intersections) == 2 and abs(discriminant) <= tangent_tol:  # If line is tangent to circle, return just one point (as both intersections have same location)
+                return [intersections[0]]
+            else:
+                return intersections
+
+    def gps_fix_cb(self, gps_msg):
+        self.is_fix_ok = True
+
+    def dubins_mission_planner(self, mission, num_points=2, inside_turn=True):
+        ''' 
+        Reads the waypoints from a MissionControl message and generates a sampled dubins 
+        path between them. 
+
+        :param mission: the original MissionControl message
+        :param turn_radius: turning radius of the robot [m]
+        :param num_points: the amount of waypoints to keep on each segment between waypoints
+        :param inside_turn: cut the waypoints corners 
+        :param int_radius: only used if inside_turn is True. Radius of the circle used to compute the intersection waypoints [m]
+        :return MissionControl.msg: new MissionControl message equal to the input one except for the waypoints attribute
+        '''
+
+        # Only continue if we have the gps fix
+        self.is_fix_ok = False
+        rospy.Subscriber("/lolo/core/gps", NavSatFix, self.gps_fix_cb)
+        while not self.is_fix_ok:
+            rospy.loginfo("Waiting for GPS fix")
+            rospy.sleep(1.0)
+        rospy.loginfo("GPS fix received")
+
+        rospy.loginfo("Computing dubins path")
+
+        turn_radius = self.bb.get(bb_enums.TURNING_RADIUS)
+        int_radius = self.bb.get(bb_enums.INTERSECTION_RADIUS)
+
+        ll_to_utm_serv = self._get_latlon_to_utm_service()
+
+        # Wait for LoLo's first position before generating the path
```

Comment:
also please avoid using robot-specific names in shared code, makes grepping harder~